tenant/src/main/scala/com/improving/app/tenant/domain/Tenant.scala

```diff
@@ -142,178 +149,146 @@ object Tenant {
     }
   }
 
-  private def updateMetaInfo(metaInfo: TenantMetaInfo, lastUpdatedByOpt: Option[MemberId]): TenantMetaInfo = {
-    metaInfo.copy(lastUpdatedBy = lastUpdatedByOpt, lastUpdated = Some(Timestamp(Instant.now())))
+  private def updateMetaInfo(metaInfo: TenantMetaInfo, lastUpdatedBy: MemberId): TenantMetaInfo = {
+    metaInfo.copy(lastUpdatedBy = lastUpdatedBy, lastUpdated = Timestamp(Instant.now()))
   }
 
   private def establishTenant(establishTenant: EstablishTenant): Either[Error, TenantEnvelope] = {
-    val maybeValidationError = applyAllValidators[EstablishTenant](
-      c => required("tenant id", tenantIdValidator)(c.tenantId),
-      c => required("activating user", memberIdValidator)(c.establishingUser)
```

Comment:
I don't think we can get away from validating stuff in scala here. I know that our existing validation is weak-sauce but it stands in for something quite a bit stronger: that these ids are not only syntactically valid but either refer to an authorized user (authorization) or that there is not already a tenant with that id

Reply:
Authorization should all be handled by gateway service, and the fact that this function takes place within an actor means we are assured that if a new tenantID comes in it will go to a new actor. In other words, the stronger things you mentioned are now part of the scope of this function.

Reply:
I guess it's okay to remove this, but I won't be surprised if some validation gets added back later